pkg/controller/kogitoapp/resource/create_resources.go

```diff
@@ -140,6 +159,62 @@ func imageStreamBuilder(chain *builderChain) *builderChain {
 	return callPostCreate(created, is, chain)
 }
 
+// Assure that all Kogito images are available before the build.
+func kogitoImageStreamBuilder(chain *builderChain) *builderChain {
+	created := true
+	var need2create []string
+
+	ocp := openshift.ImageStreamC(chain.Context.Client)
+
+	for _, key := range ImageStreamNameList {
+		isTag, _ := ocp.FetchTag(types.NamespacedName{Name: key, Namespace: ImageStreamNamespace}, ImageStreamTag)
+		if isTag == nil {
+			need2create = append(need2create, key)
+		}
+	}
+
+	if len(need2create) > 0 {
+		imageStreamLocation := fmt.Sprintf(ImageStreamLocation, ImageStreamTag+"/1")
+		log.Infof("Using imagestream %s", imageStreamLocation)
+		resp, err := http.Get(imageStreamLocation)
+		if err != nil {
+			log.Errorf("Failed to fetch imagestream %s", err.Error())
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == http.StatusOK {
+
+			var imagestreamlist imgv1.ImageStreamList
+			bodyBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				log.Warnf("Failed to read response: %s", err.Error())
+			}
+
+			if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(bodyBytes), len([]byte(bodyBytes))).Decode(&imagestreamlist); err != nil {
+				log.Errorf("Error while unmarshal file '%s': %s ", resp.Body, err)
+			}
+
+			for _, is2create := range need2create {
+				for _, is := range imagestreamlist.Items {
+					if is2create == is.Name {
+						is.Namespace = ImageStreamNamespace
```

Comment:
If I understand it correctly the operator will create image streams in openshift namespace. Does operator have privileges to do so?

Reply:
in this case yes.
I rewrote this logic, it will look, initially, for the image streams on the openshift and current namespace.
If a specific imagestream is not found on openshift and current namespace, then create it on the currenct namespace only.